RELEASES.md

```diff
@@ -26,3 +26,23 @@ The Harbor project maintains release branches for the three most recent minor re
 
 ### Next Release
 The activity for next release will be tracked in the [up-to-date project board](https://github.com/orgs/goharbor/projects/1). If your issue is not present in the corresponding release, please reach out to the maintainers to add the issue to the project board.
+
+### Publishing a New Release
+
+The following steps outline what to do when its time to plan for and publish a release. Depending on the release (major/minor/patch), not all the following items are needed.
```

Comment:
Can we be more specific about what is excluded from say a patch that needs to be included in a minor?